web/src/labels.ts

```diff
@@ -46,3 +64,89 @@ export function getUiLabels({ labels }: UILabels): string[] {
 export function orderLabels(labels: string[]) {
   return [...labels].sort((a, b) => a.localeCompare(b))
 }
+
+// This helper function takes a template type for the resources
+// and a label accessor function
+export function resourcesHaveLabels<T>(
+  features: Features,
+  resources: T[] | undefined,
+  getLabels: (resource: T) => string[]
+): boolean {
+  // Labels on resources are ignored if feature is not enabled
+  if (!features.isEnabled(Flag.Labels)) {
+    return false
+  }
+
+  if (resources === undefined) {
+    return false
+  }
+
+  return resources.some((r) => getLabels(r).length > 0)
+}
+
+// Shared resource grouping styles
+// TODO: Consider moving these to a resource grouping specific file
```

Comment:
Any opinions here? I don't mind keeping the resource grouping + label together, but don't want it to be confusing.